src/main/java/be/seeseemelk/mockbukkit/scheduler/BukkitSchedulerMock.java

```diff
@@ -366,50 +401,157 @@ public void run()
 			{
 				asyncException.set(t);
 			}
-			asyncTasksRunning.decrementAndGet();
 		}
 
 	}
 
 	@Override
-	public void runTask(Plugin plugin, Consumer<BukkitTask> task)
-	{
+	public void runTask(@NotNull Plugin plugin, @NotNull Consumer<BukkitTask> task) {
 		// TODO Auto-generated method stub
 		throw new UnimplementedOperationException();
 	}
 
 	@Override
-	public void runTaskAsynchronously(Plugin plugin, Consumer<BukkitTask> task)
+	public void runTaskAsynchronously(@NotNull Plugin plugin, @NotNull Consumer<BukkitTask> task)
 	{
 		// TODO Auto-generated method stub
 		throw new UnimplementedOperationException();
 	}
 
 	@Override
-	public void runTaskLater(Plugin plugin, Consumer<BukkitTask> task, long delay)
+	public void runTaskLater(@NotNull Plugin plugin, @NotNull Consumer<BukkitTask> task, long delay)
 	{
 		// TODO Auto-generated method stub
 		throw new UnimplementedOperationException();
 	}
 
 	@Override
-	public void runTaskLaterAsynchronously(Plugin plugin, Consumer<BukkitTask> task, long delay)
+	public void runTaskLaterAsynchronously(@NotNull Plugin plugin, @NotNull Consumer<BukkitTask> task, long delay)
 	{
 		// TODO Auto-generated method stub
 		throw new UnimplementedOperationException();
 	}
 
 	@Override
-	public void runTaskTimer(Plugin plugin, Consumer<BukkitTask> task, long delay, long period)
+	public void runTaskTimer(@NotNull Plugin plugin, @NotNull Consumer<BukkitTask> task, long delay, long period)
 	{
 		// TODO Auto-generated method stub
 		throw new UnimplementedOperationException();
 	}
 
 	@Override
-	public void runTaskTimerAsynchronously(Plugin plugin, Consumer<BukkitTask> task, long delay, long period)
+	public void runTaskTimerAsynchronously(@NotNull Plugin plugin, @NotNull Consumer<BukkitTask> task, long delay, long period)
 	{
 		// TODO Auto-generated method stub
 		throw new UnimplementedOperationException();
 	}
+
+	protected int getActiveRunningCount()
+	{
+		return pool.getActiveCount();
+	}
+
+	private static class TaskList {
+
+		private final Map<Integer, ScheduledTask> tasks;
+		private final ReadWriteLock taskLock;
+
+		private TaskList()
+		{
+			tasks = new HashMap<>();
```

Comment:
```suggestion
			tasks = new ConcurrentHashMap<>();
```
Consider using `ConcurrentHashMap` instead of `ReadWriteLock`s, in this case locks may slow down the process compared to `java.util.concurrent` API. Always avoid using low level API if possible